internal/services/databricks/databricks_virtual_network_peering_resource.go

```diff
@@ -0,0 +1,356 @@
+		d.Set("use_remote_gateways", model.Properties.UseRemoteGateways)
+
+		addressSpacePrefixes := make([]string, 0)
+		if model.Properties.DatabricksAddressSpace != nil && model.Properties.DatabricksAddressSpace.AddressPrefixes != nil {
+			addressSpacePrefixes = *model.Properties.DatabricksAddressSpace.AddressPrefixes
+		}
+		d.Set("address_space_prefixes", addressSpacePrefixes)
+
+		remoteAddressSpacePrefixes := make([]string, 0)
+		if model.Properties.RemoteAddressSpace != nil && model.Properties.RemoteAddressSpace.AddressPrefixes != nil {
+			remoteAddressSpacePrefixes = *model.Properties.RemoteAddressSpace.AddressPrefixes
+		}
+		d.Set("remote_address_space_prefixes", remoteAddressSpacePrefixes)
+
+		databricksVirtualNetworkId := ""
+		if model.Properties.DatabricksVirtualNetwork != nil && model.Properties.DatabricksVirtualNetwork.Id != nil {
+			databricksVirtualNetworkId = *model.Properties.DatabricksVirtualNetwork.Id
+		}
+		d.Set("virtual_network_id", databricksVirtualNetworkId)
+
+		remoteVirtualNetworkId := ""
+		if model.Properties.RemoteVirtualNetwork.Id != nil {
+			remoteVirtualNetworkId = *model.Properties.RemoteVirtualNetwork.Id
+		}
+		d.Set("remote_virtual_network_id", remoteVirtualNetworkId)
+	}
+
+	return nil
+}
+
+func resourceDatabricksVirtualNetworkPeeringUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataBricks.VnetPeeringClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Azure ARM databricks virtual network peering update.")
+
+	id, err := vnetpeering.ParseVirtualNetworkPeeringID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	existing, err := client.Get(ctx, *id)
+	if err != nil {
+		return fmt.Errorf("retrieving Databricks %s: %+v", *id, err)
+	}
+
+	// these are the only updatable values, so everything else in the existing.Model should still be unchanged...
+	if d.HasChange("allow_forwarded_traffic") {
+		existing.Model.Properties.AllowForwardedTraffic = pointer.To(d.Get("allow_forwarded_traffic").(bool))
+	}
+
+	if d.HasChange("allow_gateway_transit") {
+		existing.Model.Properties.AllowGatewayTransit = pointer.To(d.Get("allow_gateway_transit").(bool))
+	}
+
+	if d.HasChange("allow_virtual_network_access") {
+		existing.Model.Properties.AllowVirtualNetworkAccess = pointer.To(d.Get("allow_virtual_network_access").(bool))
+	}
+
+	if d.HasChange("use_remote_gateways") {
+		existing.Model.Properties.UseRemoteGateways = pointer.To(d.Get("use_remote_gateways").(bool))
+	}
+
+	locks.ByID(databricksVnetPeeringsResourceType)
+	defer locks.UnlockByID(databricksVnetPeeringsResourceType)
+
+	future, err := client.CreateOrUpdate(ctx, *id, *existing.Model)
+	if err != nil {
+		return fmt.Errorf("updating Databricks %s: %+v", *id, err)
+	}
+
+	if err := future.Poller.PollUntilDone(ctx); err != nil {
+		return fmt.Errorf("waiting for update of Databricks %s: %+v", *id, err)
+	}
```

Comment:
_minor_ but FYI this can be reduced too:

```suggestion
	if err := client.CreateOrUpdateThenPoll(ctx, *id, *existing.Model); err != nil {
		return fmt.Errorf("updating Databricks %s: %+v", *id, err)
	}
```

Reply:
Fixed.